### DISCUSSION TITLE
implement get_user_profile

### DISCUSSION THREAD

**Discussion by stevensu1977** [frankie567/httpx-oauth#312] [Ideas]
httpx-oauth is very useful project for me,   do we have any plan to implement get_user_profile ?

**Comment by frankie567** [comment]
Not currently. The problem with user profiles is that they vary a lot between providers, and the use-cases also.

I prefer to let developers come up with their own solution. Once you have an access token, it's usually quite easy to make the API request you need.

**Comment by iron3oxide** [comment]
Would it be possible though to provide a generic `get_userinfo()` method that simply returns whatever the given access token has in scope? I find myself having to copy paste almost all of the `get_id_email()` implementation sans the filter in the end.

**Comment by frankie567** [comment]
Done in [`0.16.0`](https://github.com/frankie567/httpx-oauth/releases/tag/v0.16.0): `get_profile` will return you the raw information from the user profile endpoint of each provider.
